state/execution.go

```diff
@@ -113,6 +113,34 @@ func (blockExec *BlockExecutor) CreateProposalBlock(
 	return state.MakeBlock(height, txs, commit, evidence, proposerAddr)
 }
 
+// ProcessProposal calls ABCI.ProcessProposal, which allows the application
+// to check if it thinks the block is valid, and add any evidence to the
+// local evidence pool in the event of fraudulent activity.
+//
+// This method returns whether the application deems this block valid.
+// TODO: In the future, pending spec work, this may become an async method.
+func (blockExec *BlockExecutor) ProcessProposal(
+	block *types.Block,
+) (bool, error) {
+	ctx := context.Background()
+	req := abci.RequestProcessProposal{
+		Txs:    block.Data.Txs.ToSliceOfBytes(),
+		Header: *block.Header.ToProto(),
+	}
+
+	resp, err := blockExec.proxyApp.ProcessProposalSync(ctx, req)
+	if err != nil {
+		return false, ErrInvalidBlock(err)
+	}
+
+	// TODO: Need to implement deserialization of evidence bytes
+	// for _, ev := range resp.Evidence {
+	// 	blockExec.evpool.AddEvidence()
+	// }
```

Comment:
The godoc states we process evidence by adding it to the evidence pool. Do we actually do that?

Reply:
Don't we want `abci.ResponseProcessProposal` to return `[]types.Evidence` instead of `[][]byte`

Reply:
More broadly, how is Tendermint expected to process application evidence? When it receives it from the application here does it just trust that it is valid. When it receives application evidence from another node does it run something like `CheckEvidence()` to make the application validate it.